lib/rubocop/minitest/support.rb

```diff
@@ -1,10 +1,269 @@
+    # underneath each character that should be highlighted, and follow
+    # the carets with a string (separated by a space) that is the
+    # message of the offense. You can include multiple offenses in
+    # one code snippet.
+    #
+    # @example Usage
+    #
+    #   class AssertNilTest < RuboCop::Minitest::Test
+    #     test "fails when using assert_equal(nil, ...)" do
+    #       assert_offense(<<~RUBY)
+    #         class FooTest < Minitest::Test
+    #           def test_do_something
+    #             assert_equal(nil, somestuff)
+    #             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `assert_nil(somestuff)`.
+    #           end
+    #         end
+    #       RUBY
+    #     end
+    #   end
+    #
+    # Autocorrection can be tested using `assert_correction` after
+    # `assert_offense`.
+    #
+    # @example `assert_offense` and `assert_correction`
+    #
+    #   class AssertNilTest < RuboCop::Minitest::Test
+    #      test "autocorrects  when using assert_equal(nil, ...)" do
+    #        assert_offense(<<~RUBY)
+    #          class FooTest < Minitest::Test
+    #           def test_do_something
+    #             assert_equal(nil, somestuff)
+    #             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `assert_nil(somestuff)`.
+    #           end
+    #         RUBY
+    #
+    #         assert_correction(<<~RUBY)
+    #           class FooTest < Minitest::Test
+    #             def test_do_something
+    #               assert_nil(somestuff)
+    #             end
+    #           end
+    #         RUBY
+    #       end
+    #     end
+    #   end
+    #
+    # If you do not want to specify an offense then use the
+    # companion method `assert_no_offenses`. This method is a much
+    # simpler assertion since it just inspects the source and checks
+    # that there were no offenses. The `assert_offense` method has
+    # to do more work by parsing out lines that contain carets.
+    #
+    # If the code produces an offense that could not be autocorrected, you can
+    # use `assert_no_corrections` after `assert_offense`.
+    #
+    # @example `assert_offense` and `assert_no_corrections`
+    #
+    #   class AssertNilTest < RuboCop::Minitest::Test
+    #     test "no autocorrections"
+    #       assert_offense(<<~RUBY)
+    #         class FooTest < Minitest::Test
+    #           def test_do_something
+    #             assert_equal(nil, somestuff)
+    #             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `assert_nil(somestuff)`.
+    #           end
+    #         end
+    #       RUBY
+    #
+    #       assert_no_corrections
+    #     end
+    #   end
+    class Test < ::Minitest::Test
+      class << self
+        # Helper to define a test method using a String. Under the hood, it replaces
+        # spaces with underscores and defines the test method.
+        #
+        #   test "verify something" do
+        #     ...
+        #   end
+        def test(name, &block)
```

Comment:
Where is this method coming from? I don't think it was here before, but if it was it seems out of scope to be part of this class

Reply:
This is a new test method, I updated the PR description (I forgot to add it), it introduces [declarative testing](https://api.rubyonrails.org/classes/ActiveSupport/Testing/Declarative.html) since all the apps I've worked in have it in some form.

Instead of always writing

```ruby
def test_something_does_something
  ...
end
```

You can do

```ruby
test "something does something" do
end
```

Which in my experience is easier to read, and you can avoid confusion with `def test_my_method_with_underscores_in_the_name` and `test "my_method with underscores in the name"` which improves test readability to me

Reply:
It doesn't replace the `def test_...` definitions but works along side

Reply:
There are gems for that purpose, `minitest/spec` itself comes pretty close with its [it](https://docs.seattlerb.org/minitest/Minitest/Spec/DSL.html#method-i-it) method. If I want to use this test method I shouldn't also get methods to assert RuboCop cops. It would only be available when in that context anyways.

Reply:
OK removed it /shrug